gst/nnstreamer/tensor_filter/tensor_filter_custom_easy.c

```diff
@@ -0,0 +1,191 @@
+/**
+ * GStreamer Tensor_Filter, Customized Module, Easy Mode
+ * Copyright (C) 2019 MyungJoo Ham <myungjoo.ham@samsung.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ */
+/**
+ * @file	tensor_filter_custom_easy.c
+ * @date	24 Oct 2019
+ * @brief	Custom tensor processing interface for simple functions
+ * @see		http://github.com/nnsuite/nnstreamer
+ * @author	MyungJoo Ham <myungjoo.ham@samsung.com>
+ * @bug		No known bugs except for NYI items
+ */
+
+#include <errno.h>
+#include <glib.h>
+#include <tensor_filter_custom_easy.h>
+#include <nnstreamer_plugin_api.h>
+#include <nnstreamer_subplugin.h>
+
+static void init_filter_custom (void) __attribute__ ((constructor));
+static void fini_filter_custom (void) __attribute__ ((destructor));
+
+/**
+ * @brief internal_data
+ */
+typedef struct _internal_data
+{
+  NNS_custom_invoke func;
+  const GstTensorsInfo *in_info;
+  const GstTensorsInfo *out_info;
+  void *data; /**< The easy-filter writer's data */
+} internal_data;
+
+/**
+ * @brief The easy-filter user's data
+ */
+typedef struct
+{
+  const internal_data *model;
+} runtime_data;
+
+
+/**
+ * @brief Register the custom-easy tensor function. More info in .h
+ * @return 0 if success. -ERRNO if error.
+ */
+int
+NNS_custom_easy_register (const char *modelname,
+    NNS_custom_invoke func, void *data,
+    const GstTensorsInfo * in_info, const GstTensorsInfo * out_info)
+{
+  internal_data *ptr;
+
+  ptr = g_new0 (internal_data, 1);
+
+  if (!ptr)
+    return -ENOMEM;
+
+  ptr->func = func;
+  ptr->data = data;
+  ptr->in_info = in_info;
```

Comment:
Should there be sanity checks for these variables (`in_info`, `out_info` and `func`), as in later functions we have asserts verifying that they exist?

Reply:
Yes they are needed.

Reply:
and fixed :)